pkg/vault/test/authorizer/TimelockAuthorizer.test.ts

```diff
@@ -731,6 +734,192 @@ describe('TimelockAuthorizer', () => {
     });
   });
 
+  describe('addCanceler', () => {
+    context('when the sender is the root', () => {
+      let scheduledId = bn(0);
+      beforeEach('set sender', async () => {
+        from = root;
+      });
+
+      context('when creating canceler', () => {
+        context('for a specific scheduled execution id', () => {
+          it('can add canceler for a specific execution id', async () => {
+            expect(await authorizer.isCanceler(scheduledId, other)).to.be.false;
+
+            await authorizer.addCanceler(scheduledId, other, { from });
+
+            expect(await authorizer.isCanceler(scheduledId, other)).to.be.true;
```

Comment:
We should also test that they are not a canceler for a second, different `scheduledId`.